CHANGELOG.md

```diff
@@ -3,6 +3,89 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## 4.0.0 - Snowball [XXXX-XX-XX]
```

Comment:
I will add the right date before the merge, that way I will make sure it is correct :)